### ISSUE TITLE
PostgreSQL: parse fails when the `on` clause includes 2 conditions with an `and`

### ISSUE THREAD

**Issue by joffreyvillard** [taozhi8833998/node-sql-parser#847]
**Describe the bug**

The parser fails on the query below (the `on` clause includes 2 conditions with an `and`):

```sql
select *
from organization
JOIN payment ON organization.id = payment.organization_id and createdat = month
```

Note that both queries below are well parsed:

Only 1 condition in the `on`
```sql
select *
from organization
JOIN payment ON organization.id = payment.organization_id
```

Parentheses surrounding the condition
```sql
select *
from organization
JOIN payment ON (organization.id = payment.organization_id and createdat = month)
```

**Database Engine**
PostgreSQL.

**To Reproduce**
-the SQL that be parsed: see above
-the node-sql-parser version: 4.0.0 
-the node version: 14.15.5

**Expected behavior**

**Screenshots**

**Additional context**
 No error, and the appropriate properties being filled up in the AST.

**Comment by taozhi8833998** [comment]
@joffreyvillard fixed in `4.0.1`

**Comment by joffreyvillard** [comment]
Excellent, thanks!
